use of variadic arguments

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

type A struct {
}

func (a A) f(vals ...bool) {
        for _, v := range vals {
                fmt.Println(v)
        }
}
func main() {
        bools := []bool{true}
        a := A{}
        a.f(bools...)
}

```

Expected result:
```console
$ go run ./sample.go
true
```

Got:
```console
$ yaegi ./sample.go
/home/marc/tmp/sample.go:14:11: panic
reflect.Set: value of type []bool is not assignable to type bool
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000164400, 0x40)
        /home/marc/sdk/go1.14.1/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0001cbe10)
        /home/marc/dev/yaegi/interp/interp.go:320 +0xbc
panic(0xc95360, 0xc0001beb00)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc00014f5e0, 0xc000324f00)
        /home/marc/dev/yaegi/interp/run.go:113 +0x20c
panic(0xc95360, 0xc0001beb00)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
reflect.Value.assignTo(0xc87600, 0xc0001276e0, 0x197, 0xdc718a, 0xb, 0xc8ed60, 0x0, 0x97, 0xc000127760, 0xc87600)
        /home/marc/sdk/go1.14.1/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xc8ed60, 0xc0001309f0, 0x181, 0xc87600, 0xc0001276e0, 0x197)
        /home/marc/sdk/go1.14.1/src/reflect/value.go:1532 +0xbd
reflect.Append(0xc87600, 0xc000127720, 0x197, 0xc0001cbbf8, 0x1, 0x1, 0x203000, 0x203000, 0x203000)
        /home/marc/sdk/go1.14.1/src/reflect/value.go:2028 +0xe8
github.com/containous/yaegi/interp.call.func5(0xc00014f5e0, 0xc000161d50)
        /home/marc/dev/yaegi/interp/run.go:795 +0x9db
github.com/containous/yaegi/interp.runCfg(0xc000324f00, 0xc00014f5e0)
        /home/marc/dev/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000102c00, 0xc000324700, 0xc00014f540)
        /home/marc/dev/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000102c00, 0xc000322000, 0xd0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/marc/dev/yaegi/interp/interp.go:388 +0x41f
main.main()
        /home/marc/dev/yaegi/cmd/yaegi/yaegi.go:94 +0x548
```
Thanks to @nrwiersma for the report.

--- a/_test/variadic6.go
+++ b/_test/variadic6.go
@@ -1,0 +1,20 @@
+package main
+
+import "fmt"
+
+type A struct {
+}
+
+func (a A) f(vals ...bool) {
+	for _, v := range vals {
+		fmt.Println(v)
+	}
+}
+func main() {
+	bools := []bool{true}
+	a := A{}
+	a.f(bools...)
+}
+
+// Output:
+// true

--- a/interp/run.go
+++ b/interp/run.go
@@ -821,7 +821,11 @@ func call(n *node) {
 						d.Set(src)
 					}
 				case variadic >= 0 && i >= variadic:
-					vararg.Set(reflect.Append(vararg, v(f)))
+					if v(f).Type() == vararg.Type() {
+						vararg.Set(v(f))
+					} else {
+						vararg.Set(reflect.Append(vararg, v(f)))
+					}
 				default:
 					dest[i].Set(v(f))
 				}

